addons/account/account.py

```diff
@@ -269,15 +269,28 @@ def search(self, cr, uid, args, offset=0, limit=None, order=None,
                 order, context=context, count=count)
 
     def _get_children_and_consol(self, cr, uid, ids, context=None):
-        #this function search for all the children and all consolidated children (recursively) of the given account ids
-        ids2 = self.search(cr, uid, [('parent_id', 'child_of', ids)], context=context)
-        ids3 = []
-        for rec in self.browse(cr, uid, ids2, context=context):
-            for child in rec.child_consol_ids:
-                ids3.append(child.id)
-        if ids3:
-            ids3 = self._get_children_and_consol(cr, uid, ids3, context)
-        return ids2 + ids3
+        res = self._query_children_and_consol(cr, uid, ids, context=context)
+        if ids and not res:
+            self._parent_store_compute(cr)
+            res = self._query_children_and_consol(cr, uid, ids, context=context)
+        return res
+
+    def _query_children_and_consol(self, cr, uid, ids, context=None):
+        if isinstance(ids, (int, long)):
+            ids = [ids]
+        cr.execute(
+            """
+            SELECT child.id
+            FROM account_account parent
+            INNER JOIN account_account child
+                ON parent.parent_left <= child.parent_left
+               AND parent.parent_right >= child.parent_right
+            WHERE parent.id IN %s
+            ORDER by COALESCE(child.level, 0), child.parent_left
```

Comment:
Well... Either I use a recursive (iterative) SQL query (PG 9+), or this whole change is now pointless since we still have to loop on consolidated children. Unless someone has a better idea, I suppose this should be closed.

Left join on account_account will not get children of consolidated children.